### ISSUE TITLE
Create directory for dump-mir-dir automatically

### ISSUE THREAD

**Issue by wdv4758h** [rust-lang/rust#35543]
I have tried this command to dump MIR things : 

``` sh
cargo rustc -- -Z dump-mir=all -Z dump-mir-dir=target/mir/
```

But it won't create the directory for me, I need to create the directory before running the command.
If the directory doesn't exist, it won't show any error message and it won't generate any MIR file.

I think we should create the directory automatically. 
Even not doing so, we should show some error message. 

Addtionally, I wonder what are other people's opinions on setting default MIR dumping directory to somewhere like `target/mir/` instead of current directory.

Using : 
- rustc 1.12.0-nightly (b30eff7ba 2016-08-05)
- cargo 0.13.0-nightly (4e66190 2016-08-05)

**Comment by Mark-Simulacrum** [comment]
I'm marking this as E-easy and E-mentor; we need to add a `fs::create_dir_all` call [here](https://github.com/rust-lang/rust/blob/20de961226de41a18ca1d9b166c6bc607f3a597a/src/librustc_mir/util/pretty.rs#L107) on the parent directory of the file path. Let me know if you'd like to work on this.

**Comment by wdv4758h** [comment]
sure, I can work on this :)

**Comment by Mark-Simulacrum** [comment]
Go ahead! Please cc me on the PR when you create it, and let us know if you need any help; leaving a comment here works, or you can find help on IRC (#rust-internals on irc.mozilla.org).
